src/main/java/com/esri/samples/symbology/read_symbols_from_mobile_style_file/ReadSymbolsFromMobileStyleFileController.java

```diff
@@ -0,0 +1,353 @@
+   */
+  private void updateSymbolPreview(MultilayerPointSymbol multilayerPointSymbol) {
+    // remove the previously displayed image view
+    symbolPreviewHBox.getChildren().clear();
+
+    // create an image and image view from the symbol
+    ListenableFuture<Image> symbolImageFuture = multilayerPointSymbol.createSwatchAsync(0x00000000, 1);
+    try {
+      Image symbolImage = symbolImageFuture.get();
+      ImageView symbolImageView = new ImageView(symbolImage);
+      // display the image view in the preview area
+      symbolPreviewHBox.getChildren().add(symbolImageView);
+
+    } catch (InterruptedException | ExecutionException e) {
+      new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided MultilayerPointSymbol" + e.getMessage()).show();
+    }
+  }
+
+  /**
+   * Updates the size of the symbol preview and the label when the slider is interacted with.
+   */
+  @FXML
+  private void updateSymbolSize() {
+    // get the slider value and convert to a string
+    String sizeValue = Integer.toString((int) sizeSlider.getValue());
+    // display the value in the label
+    sizeLabel.setText(sizeValue + " px");
+
+    // update the preview
+    buildCompositeSymbol();
+  }
+
+  /**
+   * Adds graphics to the map view on mouse clicks.
+   * @param e mouse button click event
+   */
+  @FXML
+  private void handleMouseClicked(MouseEvent e){
+    if (e.getButton() == MouseButton.PRIMARY && e.isStillSincePress()) {
+      // convert clicked point to a map point
+      Point mapPoint = mapView.screenToLocation(new Point2D(e.getX(), e.getY()));
+
+      // create a new graphic with the point and symbol
+      Graphic graphic = new Graphic(mapPoint, faceSymbol);
+      graphicsOverlay.getGraphics().add(graphic);
+    }
+  }
+
+  /**
+   * Clears all the graphics from the graphics overlay.
+   */
+  @FXML
+  private void resetView() {
+    graphicsOverlay.getGraphics().clear();
+  }
+
+  /**
+   * Stops and releases all resources used in application.
+   */
+  public void terminate() {
+
+    if (mapView != null) {
+      mapView.dispose();
+    }
+  }
+
+  /**
+   * Shows the available symbol of the SymbolStyleSearchResult in the symbol selection list view.
+   */
+  class SymbolLayerInfoListCell extends ListCell<SymbolStyleSearchResult> {
+
+    private SymbolLayerInfoListCell() {
+      // set the cell to display only a graphic
+      setContentDisplay(ContentDisplay.GRAPHIC_ONLY);
+    }
+
+    protected void updateItem(SymbolStyleSearchResult item, boolean empty) {
+      super.updateItem(item, empty);
+
+      if (item == null || empty) {
+        // if the item in the list view is an empty item, show nothing
+        setGraphic(null);
+      } else {
+        // get the symbol from the list view entry, and create an image from it
+        ListenableFuture<Image> symbolImageFuture = item.getSymbol().createSwatchAsync(0x00000000, 1);
+        try {
```

Comment:
Use a DoneListener?

Reply:
As with the above, putting this in a `doneListener` seems to cause some graphical bugs in the `ListView` when selecting different symbols. Not sure whether this is a JavaFX issue or a runtime issue though.